packages/lib/server/getLuckyUser.integration-test.ts

```diff
@@ -0,0 +1,289 @@
+import { describe, it, expect, afterEach, beforeAll, afterAll } from "vitest";
+
+import prisma from "@calcom/prisma";
+
+import { getLuckyUser } from "./getLuckyUser";
+
+describe("getLuckyUser tests", () => {
+  describe("should not consider no show bookings for round robin when: ", () => {
+    let userIds: number[] = [];
+    let eventTypeId: number;
+
+    beforeAll(async () => {
+      const event = await prisma.eventType.create({
+        data: {
+          title: "Test Event",
+          slug: "test-event",
+          length: 15,
+        },
+        select: {
+          id: true,
+        },
+      });
+      eventTypeId = event.id;
+    });
+
+    afterEach(async () => {
+      await prisma.user.deleteMany({
+        where: {
+          id: {
+            in: userIds,
+          },
+        },
+      });
+
+      userIds = [];
+    });
+
+    afterAll(async () => {
+      await prisma.eventType.delete({
+        where: {
+          id: eventTypeId,
+        },
+      });
+    });
+
+    it("Host is a no show", async () => {
+      const user1 = await prisma.user.create({
+        data: {
+          email: "test-user1@example.com",
+          bookings: {
+            create: [
+              {
+                uid: "uuid-test-user1-booking1",
+                createdAt: new Date("2022-01-25T05:30:00.000Z"),
+                title: "Test user 1 Booking",
+                startTime: new Date(),
+                endTime: new Date(),
+                eventTypeId,
+                attendees: {
+                  create: [
+                    {
+                      name: "test-attendee",
+                      email: "test-attendee@example.com",
+                      timeZone: "Asia/Calcutta",
+                    },
+                  ],
+                },
+              },
+            ],
+          },
+        },
+      });
+
+      const user2 = await prisma.user.create({
+        data: {
+          email: "test-user2@example.com",
+          bookings: {
+            create: [
+              {
+                uid: "uuid-test-user2-booking1",
+                title: "Test User 2 Booking",
+                createdAt: new Date("2022-01-25T06:30:00.000Z"),
+                noShowHost: true,
+                startTime: new Date(),
+                endTime: new Date(),
+                attendees: {
+                  create: [
+                    {
+                      name: "test-attendee",
+                      email: "test-attendee@example.com",
+                      timeZone: "Asia/Calcutta",
+                    },
+                  ],
+                },
+                eventTypeId,
+              },
+            ],
+          },
+        },
+      });
+
+      userIds.push(user1.id, user2.id);
+
+      expect(
+        getLuckyUser("MAXIMIZE_AVAILABILITY", {
+          availableUsers: [user1, user2],
+          eventTypeId,
+        })
+      ).resolves.toStrictEqual(user2);
```

Comment:
This expectation is randomly failing locally.

https://www.loom.com/share/b47c9fb20d554068aab0cebadca3fc6d